web-services/client/src/main/java/datawave/webservice/query/QueryParametersImpl.java

```diff
@@ -89,11 +110,21 @@ public void validate(Map<String,List<String>> parameters) throws IllegalArgument
                 throw new IllegalArgumentException("Unknown condition.");
             }
         }
-        Preconditions.checkNotNull(this.query, "Query string cannot be null");
-        Preconditions.checkNotNull(this.queryName, "Query name cannot be null");
-        Preconditions.checkNotNull(this.persistenceMode, "Persistence mode cannot be null");
-        Preconditions.checkNotNull(this.auths, "Query auths cannot be null");
-        Preconditions.checkNotNull(this.expirationDate, "Expiration date cannot be null");
+        
+        try {
+            Preconditions.checkNotNull(this.query, "QueryParameter 'query' cannot be null");
+            Preconditions.checkNotNull(this.queryName, "QueryParameter 'queryName' cannot be null");
+            Preconditions.checkNotNull(this.persistenceMode, "QueryParameter 'persistence' mode cannot be null");
+            Preconditions.checkNotNull(this.auths, "QueryParameter 'auths' cannot be null");
+            Preconditions.checkNotNull(this.pagesize, "QueryParameter 'pagesize' cannot be null");
+            Preconditions.checkNotNull(this.pageTimeout, "QueryParameter 'pageTimeout' cannot be null");
+            Preconditions.checkNotNull(this.expirationDate, "QueryParameter 'expiration' date cannot be null");
+            Preconditions.checkNotNull(this.beginDate, "QueryParameter 'begin' cannot be null");
```

Comment:
Since it looks like EdgeQuery has optional dates, and the parsing above this block actually sets beginDate and endDate explicitly to null in some cases, I'd drop these checkNotNull's.

Reply:
I'll update this soon, thanks for pointing that out.